### ISSUE TITLE
Exceptions should not be treated any differently than any other log property

### ISSUE THREAD

**Issue by konste** [serilog/serilog#251]
I have asked about it in Serilog Google group (and repeat it below for completeness), but in short I'm very much concerned that Serilog API is doubled to treat exceptions differently than any other property type. IMHO they should be handled uniformly and in case some types (like exception) require special processing it should be done by recognizing those cases dynamically, rather than through specialized API which takes argument of specifically that type.

One specific feature of Serilog interface creates a trouble for us - having separate methods for reporting exceptions. For example:
        Log.Error ("format {propertyName}", propertyValue) and
        Log.Error (exception, "format {propertyName}", propertyValue);

There are several reasons why second variant does not work for us - for one we retrofit Serilog instead of legacy logging, which does not treat exceptions any different than other structured data. Another related reason is that we may have more than one exception object per log statement.

Long story short - we need to be able to handle statements like this: 
        Log.Error ("exception1 = {ex1}, exception2 = {ex2}", exception1, exception2); 

I have very elaborate enricher for exception objects, but unfortunately it is called only for statements where exception is called explicitly:  Log.Error (exception,...
When exception object is one of the message parameters or when exception is added as ForContext("exceptionPropertyName", exceptionPropertyObject) my enricher is not called (or actually it is, but exception object is already lost by that time).

My question is how to implement enricher or destructure policy or whatever, so that exceptions passed as message parameters or context properties get properly destructured by our code.
